ksqldb-engine/src/main/java/io/confluent/ksql/execution/pull/PullPhysicalPlan.java

```diff
@@ -79,27 +84,53 @@ public PullPhysicalPlan(
 
   public void execute(
       final List<KsqlPartitionLocation> locations,
-      final PullQueryQueue pullQueryQueue,
-      final BiFunction<List<?>, LogicalSchema, PullQueryRow> rowFactory) {
-
+      final PullQueryWriteStream pullQueryQueue,
+      final Function<StreamedRow, StreamedRow> addDebugInfo
+  ) {
     // We only know at runtime which partitions to get from which node.
     // That's why we need to set this explicitly for the dataSource operators
     dataSourceOperator.setPartitionLocations(locations);
 
+    // to achieve parity with the remote execution path, we start
+    // by passing the row header into the queue
+    pullQueryQueue.write(ImmutableList.of(StreamedRow.header(queryId, schema)));
+
     open();
-    QueryRow row;
-    while ((row = (QueryRow)next()) != null) {
-      if (pullQueryQueue.isClosed()) {
-        // If the queue has been closed, we stop adding rows and cleanup. This should be triggered
-        // because the client has closed their connection with the server before the results have
-        // completed.
-        LOGGER.info("Queue closed before results completed. Stopping execution.");
-        break;
-      }
-      if (!pullQueryQueue.acceptRow(rowFactory.apply(row.value().values(), schema))) {
-        LOGGER.info("Failed to queue row");
+    QueryRow row = (QueryRow) next();
+
+    int rowsQueued = 0;
```

Comment:
final!

Reply:
oops I deleted this altogether, I used this for debugging and it's not used anymore :)

Reply:
Checkstyle was gonna git you!

Reply:
Shockingly it didn't! I ran checkstyle and spotbugs before pushing